api/extra.go

```diff
@@ -49,14 +55,102 @@ func ResolveDidToHandle(ctx context.Context, xrpcc *xrpc.Client, res did.Resolve
 		return "", "", fmt.Errorf("our XRPC client is authed for a different pds (%s != %s)", svc.ServiceEndpoint, xrpcc.Host)
 	}
 
-	verdid, err := comatproto.IdentityResolveHandle(ctx, xrpcc, handle)
+	verdid, err := hr.ResolveHandleToDid(ctx, handle)
 	if err != nil {
 		return "", "", err
 	}
 
-	if verdid.Did != udid {
+	if verdid != udid {
 		return "", "", fmt.Errorf("pds server reported different did for claimed handle")
 	}
 
 	return handle, svc.ServiceEndpoint, nil
 }
+
+type HandleResolver interface {
+	ResolveHandleToDid(ctx context.Context, handle string) (string, error)
+}
+
+type ProdHandleResolver struct {
+}
+
+func (dr *ProdHandleResolver) ResolveHandleToDid(ctx context.Context, handle string) (string, error) {
+	c := http.DefaultClient
+
+	resp, wkerr := c.Get(fmt.Sprintf("https://%s/.well-known/atproto-did", handle))
+	if wkerr == nil && resp.StatusCode == 200 {
+		b, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return "", fmt.Errorf("failed to read resolved did: %w", err)
+		}
+
+		parsed, err := did.ParseDID(string(b))
+		if err != nil {
+			return "", err
+		}
+
+		return parsed.String(), nil
+	}
+	log.Infof("failed to resolve handle (%s) through well-known route: %s", handle, wkerr)
+
+	res, err := net.LookupTXT("_atproto." + handle)
```

Comment:
In typescript the HTTP and TXT resolution happens in parallel (using async). The consensus when chatting with Dan was to prefer DNS TXT over HTTPS when doing them sequentially, so try the DNS method before the HTTPS method. We consider the HTTPS method a fallback for large hosting providers etc.

This isn't blocking, especially if it causes the test/dev flow to break. Implementations can do whichever prioritization they want.